src/Led.cpp

```diff
@@ -224,500 +255,566 @@ static void Led_Task(void *parameter) {
-					if (showEvenError) {
-						if (Led_Address(led) % 2 == 0) {
-							if (millis() <= 10000) {
-								leds[Led_Address(led)] = CRGB::Orange;
-							} else {
-								leds[Led_Address(led)] = CRGB::Red;
-							}
-						}
-					} else {
-						if (millis() >= 10000) { // Flashes red after 10s (will remain forever if SD cannot be mounted)
-							leds[Led_Address(led)] = CRGB::Red;
-						} else {
-							if (Led_Address(led) % 2 == 1) {
-								leds[Led_Address(led)] = CRGB::Orange;
-							}
-						}
-					}
-				}
-				FastLED.show();
-				showEvenError = !showEvenError;
-				vTaskDelay(portTICK_RATE_MS * 500);
-				continue;
+
+			uint32_t taskDelay = 20;
+			
+			// check indications and set led-mode
+			if (!LED_INDICATOR_IS_SET(LedIndicatorType::BootComplete)){
+				nextAnimaiton = LedAnimationType::Boot;
+			} else if (CheckForPowerButtonAnimation()) { 
+				nextAnimaiton = LedAnimationType::Shutdown;
+			} else if (LED_INDICATOR_IS_SET(LedIndicatorType::Error)){
+				LED_INDICATOR_CLEAR(LedIndicatorType::Error);
+				nextAnimaiton = LedAnimationType::Error;
+			} else if (LED_INDICATOR_IS_SET(LedIndicatorType::Ok)) {
+				LED_INDICATOR_CLEAR(LedIndicatorType::Ok);
+				nextAnimaiton = LedAnimationType::Ok;
+			#ifdef BATTERY_MEASURE_ENABLE
+			} else if (LED_INDICATOR_IS_SET(LedIndicatorType::VoltageWarning)) {
+				LED_INDICATOR_CLEAR(LedIndicatorType::VoltageWarning);
+				nextAnimaiton = LedAnimationType::VoltageWarning;
+			} else if (LED_INDICATOR_IS_SET(LedIndicatorType::Voltage)) {
+				nextAnimaiton = LedAnimationType::BatteryMeasurement;
+			#endif
+			} else if (hlastVolume != AudioPlayer_GetCurrentVolume()) {
+				nextAnimaiton = LedAnimationType::Volume;
+			} else if (LED_INDICATOR_IS_SET(LedIndicatorType::Rewind)) {
+				LED_INDICATOR_CLEAR(LedIndicatorType::Rewind);
+				nextAnimaiton = LedAnimationType::Rewind;
+			} else if (LED_INDICATOR_IS_SET(LedIndicatorType::PlaylistProgress)){
+				nextAnimaiton = LedAnimationType::Playlist;
+			} else if (gPlayProperties.playlistFinished) {
+				nextAnimaiton = LedAnimationType::Idle; // todo: what is valid?
+			} else if (gPlayProperties.currentSpeechActive) {
+				nextAnimaiton = LedAnimationType::Speech;
+			} else if (gPlayProperties.pausePlay) {
+				nextAnimaiton = LedAnimationType::Pause;
+			} else if (gPlayProperties.isWebstream) {
+				nextAnimaiton = LedAnimationType::Webstream;
+			} else if ((gPlayProperties.playMode != BUSY) && (gPlayProperties.playMode != NO_PLAYLIST)) {
+				nextAnimaiton = LedAnimationType::Progress;
+			} else if (gPlayProperties.playMode == NO_PLAYLIST){
+				nextAnimaiton = LedAnimationType::Idle;
+			} else if (gPlayProperties.playMode == BUSY) {
+				nextAnimaiton = LedAnimationType::Busy;
+			} else {
+				nextAnimaiton = LedAnimationType::NoNewAnimation; // should not happen
 			}
 
+			// check for instant transition
+			if ((activeAnnimation != nextAnimaiton) && (nextAnimaiton < activeAnnimation)) {
```

Comment:
Yo can remove the check for not-equal, since if the "nextAnimaiton < activeAnnimation" is true, then they are not equal anyways :).

```suggestion
			if (nextAnimaiton < activeAnnimation) {
```

Reply:
Oh that's stupid ^^. Thanks